app/src/main/java/com/mapbox/maps/testapp/utils/NavigationSimulator.kt

```diff
@@ -338,12 +333,10 @@ class NavigationSimulator(
   }
 
   override fun onIndicatorPositionChanged(point: Point) {
-    val progress = TurfMeasurement.length(
-      TurfMisc.lineSlice(routeStartPoint, point, routePoints),
-      TurfConstants.UNIT_CENTIMETERS
-    ) / totalRouteLength
-    routeLayer.lineTrimOffset(listOf(0.0, progress))
-    casingLayer.lineTrimOffset(listOf(0.0, progress))
+    // use altitude to pass through interpolated progress data, reduce the overhead to
```

Comment:
Won't it actually lead to the route line being slightly out of sync with the puck when it's animating between location updates?

Reply:
Discussed - I've missed the point that altitude is animated together with the lat-lon, thus line trim will be updated correctly